argo/src/Argo.hs

```diff
@@ -851,15 +864,17 @@ handleRequest ::
   Request ->
   IO ()
 handleRequest opts respond app req = do
-  case M.lookup method $ appMethods app of
-    Nothing -> throwIO $ (methodNotFound method) { errorID = reqID }
+  case Map.lookup method $ appMethods app of
+    Nothing -> do
+      throwIO $ (methodNotFound method) { errorID = reqID }
     Just (CommandMethod m) -> withActiveThread app $ withRequestID $ \reqID -> do
       stateID <- getStateID req
       response <- withMVar theState $ \state -> do
-        when (stateID == initialStateID) $ do
-          stateCount <- statePoolCount state
-          when (stateCount >= (optMaxOccupancy opts)) $
-            throwIO serverAtCapacity
+        serverFull <- do cnt <- statePoolCount state
+                         pure $ cnt >= (optMaxOccupancy opts)
+                                && stateID == initialStateID
```

Comment:
Why do we only consider the server full if the request's state ID is the initial state ID? Wouldn't this mean that the server would only ever be considered full if `optMaxOccupancy == 1`?

Reply:
If a command is made from some non-initial state `X` and it creates some new non-initial state `Y`, `X` is destroyed and replaced with `Y` (as things are currently implemented -- historically this has not always been the case). So the number of live states is maintained. We only get an _additional_ state if it's a fresh request, since we always keep around the initial state.

I'll add a quick note to clarify.